grid/social-media/followers/profile.supr

```diff
@@ -51,6 +51,32 @@ usecase GetFollowers safe {
       """
       imageUrl string
     } ]
+    """
+    Rate limit
+    Rate limit details.
+    """
+    rateLimit {
```

Comment:
Harmonization of rate limiting across providers will be hard. Some of them work with rolling window, some do not provide exact limit number.

Reply:
I see the biggest challenge with use-cases performing multiple requests in a single perform _and_ different limits per endpoint.

Reply:
> I see the biggest challenge with use-cases performing multiple requests in a single perform and different limits per endpoint.

I see two options:

1. Instead of returning remaining requests send to the client remaining performs of the use-case and calculate the worst case.

But I do not know what to do in case of subsequent fetches of referenced entities (e. g. attachments) with variable count.

2. Return remaining requests in array of buckets (brings complexity to the client)

```
rateLimits: [
 {
  bucket: 'followers',
  requestsPerUsecase: 1,
  remainingRequestsAvailable: 100,
  ...
 },
 {
  bucket: 'attachments',
  requestsPerUsecase: undefined, //we do not know the number
  remainingRequestsAvailable: 10,
  ...
 }
]
```